ui/pages/confirmations/components/confirm/info/approve/hooks/use-approve-token-simulation.ts

```diff
@@ -0,0 +1,36 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useSelector } from 'react-redux';
+import { getIntlLocale } from '../../../../../../../ducks/locale/locale';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { useDecodedTransactionData } from '../../hooks/useDecodedTransactionData';
+import { useIsNFT } from './use-is-nft';
+
+const UNLIMITED_THRESHOLD = 10 ** 15;
```

Comment:
This is a difficult decision to make, as theoretically someone could want to approve `(10 ** 18) - 1`?

Reply:
I've changed the threshold to be derived from the number of decimals

Reply:
The circulating token supply is stored in a uint256 so I believe the theoretical maximum is much higher. However, the decision was to consider the cap UNLIMITED (with the true value still visible in a tooltip) at 10 ** 15 because this is the threshold at which we use ellipsis throughout the application code.